### ISSUE TITLE
Adapt guide to be inclusive of API testing

### ISSUE THREAD

**Issue by itscooper** [OWASP/OWASP-Testing-Guide-v5#5]
Web-based APIs can, for the most part, be testing using this guide. Some elements are naturally client-side and therefore irrelevant to API testing.

I suggest that we create an article in section 3 or 4.1 that talks about how the guide can be used for API testing.

I further suggest that we review the existing articles to ensure that the language used is appropriate for web app and APIs alike, and determine where any additions might be needed where testing for the same issue on an API involves a different process.

**Comment by victoriadrake** [comment]
#76 partially addressed this issue. A review of existing articles is still outstanding.

**Comment by garthoid** [comment]
I see there is “testing for APIs” page. I would like to suggest taking this content and reworking it into several subtopics within the API section.

Starting with a 00-API Testing Overview Section.
Then renaming  01-Testing for GraphQL to some later number but keeping the content intact
Next, 01-Finding APIs, the recon of finding APIs to test
Following with section on specific testing based on the API top 10.

Interesting in leadership thoughts on this approach.

**Comment by victoriadrake** [comment]
> I see there is “testing for APIs” page (in the root of the repo). I would like to suggest taking this content and reworking it into several subtopics within the API section.
> 
> Starting with a 00-API Testing Overview Section. Then renaming 01-Testing for GraphQL to some later number but keeping the content intact Next, 01-Finding APIs, the recon of finding APIs to test Following with section on specific testing based on the API top 10.
> 
> Interesting in leadership thoughts on this approach.

I think this makes sense! Would you like to do a rough draft for the new content? I'm happy to help go over it then.

**Comment by garthoid** [comment]
> I think this makes sense! Would you like to do a rough draft for the new content? I'm happy to help go over it then.

Thank you. I almost have a PR ready for the Overview section.

**Comment by garthoid** [comment]
> I think this makes sense! Would you like to do a rough draft for the new content? I'm happy to help go over it then.

@victoriadrake There is a PR available for you to review.

**Comment by irgoncalves** [comment]
Hi @garthoid , @kingthorin , @victoriadrake , team, if I am reading this correctly, is the goal to have under 'API Testing' section specific test cases based on the Top 10 such as 'Testing for BOLA',  'Testing for Rate Limiting' etc? If so, would each of them be added as a separate issue folks can contribute other than this one?

**Comment by kingthorin** [comment]
IMHO only if it's something that's truly unique to API testing. From my PoV BOLA is essentially IDOR by a different name, but in reality is likely tested the same as IDOR or as any other authz issue.

**Comment by irgoncalves** [comment]
Thank you @kingthorin for the input. Yeah, I don't see a substantial difference between IDOR and BOLA as well, but I view BOLA as a more specific term that has gained traction in the context of APIs, especially due to the OWASP Top 10 API. I believe the current WSTG-AUTHZ-04 could benefit from being more inclusive of API testing. I have two proposals that I'd like to discuss and I can make contributions:
1) Adapt/enhance WSTG-AUTHZ-04 to include API-specific considerations for testing for BOLA and potentially rename it to reflect this inclusion.
2) Create a new BOLA test case scenario under API Testing section. This would be focused explicitly on API object-level authorization issues and goes aligned with comments on [here](https://github.com/OWASP/wstg/issues/5#issuecomment-2143800909).

**Comment by kingthorin** [comment]
I think option 2.

**Comment by garthoid** [comment]
I would agree with option 2, with any relevant references to WSTG-AUTHZ-04.

**Comment by irgoncalves** [comment]
Sounds good. I will work on option 2 and send a PR in a few days referencing this issue.
